Microsoft.Build.Sql: does not work with NuGet centralized package management

<!-- ⚠️⚠️ Do Not Delete This! bug_report_template ⚠️⚠️ -->
<!-- Please read our Rules of Conduct: https://opensource.microsoft.com/codeofconduct/ -->
<!-- Please search existing issues to avoid creating duplicates. -->
<!-- Also please test using the latest release to make sure your issue has not already been fixed. -->
<!-- To obtain your SqlPackage version, use sqlpackage.exe /Version -->

- SqlPackage or DacFx Version: 0.1.10-preview
- .NET Framework (Windows-only) or .NET Core: .NET 7.0
- Environment (local platform and source/target platforms): Windows, GitHub Actions on Ubuntu / Windows agents

**Steps to Reproduce:**

Using the following config in `Directory.Packages.props`:

```xml
<Project>
  <PropertyGroup>
    <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
  </PropertyGroup>
  <!-- SQL Build Tools -->
  <ItemGroup>
    <PackageVersion Include="Microsoft.Build.Sql" Version="0.1.10-preview" />
  </ItemGroup>
</Project>
```

And the following `.sqlproj` schema, created using the `dotnet new sqlproj` template:

```xml
<?xml version="1.0" encoding="utf-8"?>
<Project DefaultTargets="Build">
  <Sdk Name="Microsoft.Build.Sql"/>
  <PropertyGroup>
    <Name>Sdkbin.Database</Name>
    <DSP>Microsoft.Data.Tools.Schema.Sql.Sql160DatabaseSchemaProvider</DSP>
    <ModelCollation>1033, CI</ModelCollation>
  </PropertyGroup>
</Project>
```

And the following `global.json`:

```json
{
  "sdk": {
    "rollForward": "latestMinor",
    "version": "7.0.100"
  },
   "msbuild-sdks": {
        "Microsoft.Build.Sql": "0.1.10-preview"
    }
}
```

Running `dotnet build -c Release` produces the following error:

```
C:\Repositories\project\MyApp.Database.sqlproj : error NU1008: Projects tha t use central package version management should not define the version on the PackageReference items but on the Package Version items: Microsoft.NETFramework.ReferenceAssemblies.
    0 Warning(s)
    1 Error(s)
```

I tried working around this by updating `Directory.Packages.props` to:

```xml
<Project>
  <PropertyGroup>
    <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
  </PropertyGroup>
  <!-- SQL Build Tools -->
  <ItemGroup>
    <PackageVersion Include="Microsoft.Build.Sql" Version="0.1.10-preview" />
    <PackageVersion Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.3" />
  </ItemGroup>
</Project>
```

But I still get the same build error.

**Did this occur in prior versions?  If not - which version(s) did it work in?**

Haven't used this tool before and this is a greenfield project, so I don't know.

--- a/src/Microsoft.Build.Sql/sdk/Sdk.targets
+++ b/src/Microsoft.Build.Sql/sdk/Sdk.targets
@@ -33,7 +33,7 @@
 
   <ItemGroup>
     <!-- This is necessary for building on non-Windows platforms. -->
-    <PackageReference Condition="'$(NetCoreBuild)' == 'true'" Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.0" PrivateAssets="All" />
+    <PackageReference Condition="'$(NetCoreBuild)' == 'true'" Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.0" PrivateAssets="All" IsImplicitlyDefined="true" />
   </ItemGroup>
 
   <ItemGroup>

--- a/test/Microsoft.Build.Sql.Tests/BuildTests.cs
+++ b/test/Microsoft.Build.Sql.Tests/BuildTests.cs
@@ -206,5 +206,23 @@ namespace Microsoft.Build.Sql.Tests
                 }
             }
         }
+
+        [Test]
+        [Description("Issue #316: Verifies build with central package management turned on.")]
+        public void VerifyBuildWithCentralPackageManagement()
+        {
+            ProjectUtils.AddProperties(this.GetProjectFilePath(), new Dictionary<string, string>()
+            {
+                { "ManagePackageVersionsCentrally", "True" }
+            });
+
+            string stdOutput, stdError;
+            int exitCode = this.RunDotnetCommandOnProject("build", out stdOutput, out stdError);
+
+            // Verify success
+            Assert.AreEqual(0, exitCode, "Build failed with error " + stdError);
+            Assert.AreEqual(string.Empty, stdError);
+            this.VerifyDacPackage();
+        }
     }
 }

